browser/ui/sidebar/sidebar_web_panel_controller.cc

```diff
@@ -7,23 +7,48 @@
 
 #include "brave/browser/ui/sidebar/sidebar_utils.h"
 #include "brave/browser/ui/views/frame/split_view/brave_multi_contents_view.h"
+#include "chrome/browser/ui/browser_tabstrip.h"
 #include "chrome/browser/ui/views/frame/browser_view.h"
 
 namespace sidebar {
 
 SidebarWebPanelController::SidebarWebPanelController(BrowserView& browser_view)
     : browser_view_(browser_view) {
   CHECK(IsWebPanelFeatureEnabled());
+  browser_view_->browser()->tab_strip_model()->AddObserver(this);
 }
 
-SidebarWebPanelController::~SidebarWebPanelController() = default;
+SidebarWebPanelController::~SidebarWebPanelController() {
+  CHECK(!panel_contents_);
+}
+
+void SidebarWebPanelController::ToggleWebPanel(const SidebarItem& item) {
+  // If panel is for |item|, close and return.
+  // Otherwise, open new panel after closing.
```

Comment:
Maybe noting an edge case this logic is covering would be good

```
When item is same but url has been chaned, closes the current panel and reopens the panel with the new url.
```

Reply:
Added.